### ISSUE TITLE
Python 3 LineTooLong raised by moto S3

### ISSUE THREAD

**Issue by tmylk** [gabrielfalcao/HTTPretty#291]
Python 3 is not supported but I am trying to fix this issue and would appreciate advice.

After upgrading to  httpretty==0.8.14 from httpretty==0.8.10 LineTooLong exception [appeared in Python 3 in smart_open using moto](https://travis-ci.org/piskvorky/smart_open/jobs/119058195). (in 0.8.12 there are other issues and tests don't run.)

It is raised by the innocuous lines in `lib/python3.4/http/client.py` where fp is a `FakeSockFile`, `len(line)=65537, _MAXLINE=65536`

``` python
        line = fp.readline(_MAXLINE + 1)
        if len(line) > _MAXLINE:
            raise LineTooLong("header line")
```

Exception:

```
  File "/opt/python/3.3.5/lib/python3.3/threading.py", line 901, in _bootstrap_inner
    self.run()
  File "/opt/python/3.3.5/lib/python3.3/threading.py", line 858, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/python/3.3.5/lib/python3.3/multiprocessing/pool.py", line 347, in _handle_tasks
    for i, task in enumerate(taskseq):
  File "/opt/python/3.3.5/lib/python3.3/multiprocessing/pool.py", line 274, in <genexpr>
    self._taskqueue.put((((result._job, i, func, (x,), {})
  File "/home/travis/build/piskvorky/smart_open/smart_open/smart_open_lib.py", line 636, in <genexpr>
    keys = (key for key in bucket.list(prefix=prefix) if accept_key(key.name))
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/s3/bucketlistresultset.py", line 34, in bucket_lister
    encoding_type=encoding_type)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/s3/bucket.py", line 473, in get_all_keys
    '', headers, **params)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/s3/bucket.py", line 399, in _get_all
    query_args=query_args)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/s3/connection.py", line 665, in make_request
    retry_handler=retry_handler
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/connection.py", line 1071, in make_request
    retry_handler=retry_handler)
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/connection.py", line 1030, in _mexe
    raise ex
  File "/home/travis/virtualenv/python3.3.5/lib/python3.3/site-packages/boto/connection.py", line 944, in _mexe
    response = connection.getresponse()
  File "/opt/python/3.3.5/lib/python3.3/http/client.py", line 1147, in getresponse
    response.begin()
  File "/opt/python/3.3.5/lib/python3.3/http/client.py", line 382, in begin
    self.headers = self.msg = parse_headers(self.fp)
  File "/opt/python/3.3.5/lib/python3.3/http/client.py", line 264, in parse_headers
    raise LineTooLong("header line")
``
```

**Comment by mikedlr** [comment]
@tmylk

Is this issue still live?  Do you have / could you create a test case which would demonstrate it in httpretty?  There seems to be some interest in working on httpretty for Python 3 (see #307 )

**Comment by sduenas** [comment]
I had the same problem and I found that this bug was fixed by 7935ab4bd3288129136fbc890ba5e55a8f34609f.

**Comment by tmylk** [comment]
The issue is fixed with `httpretty=0.8.14` and `moto==0.4.30`. Tested locally.
However `moto` has pinned `httpretty=0.8.10` so we can't use it in CI.
The reason it is pinned is the lack of Python 3 support (see https://github.com/spulec/moto/issues/514 )
